metricbeat/module/kubernetes/apiserver/apiserver_test.go

```diff
@@ -0,0 +1,23 @@
+// +build !integration
+
+package apiserver
+
+import (
+	"testing"
+
+	"github.com/elastic/beats/metricbeat/helper/prometheus"
+)
+
+const testFile = "_meta/test/metrics"
+
+func TestEventMapping(t *testing.T) {
+	prometheus.TestMetricSet(t, "kubernetes", "apiserver",
+		prometheus.TestCases{
+			{
+				MetricsFile:  "./_meta/test/metrics",
+				ExpectedFile: "./_meta/test/metrics.expected",
```

Comment:
This reminds me a lot of what @simitt was doing here with the approvals: https://github.com/elastic/apm-server/blob/master/tests/approvals.go I wanted to look into it to also do some comparison on json docs for Elasticsearch. Perhaps we can sync up on this.

Reply:
Thats the `approvals` script that you can run from cmd line: https://github.com/elastic/apm-server/blob/master/tests/scripts/approvals.go

Would be great to put this in its own small library for more generic usage.

Reply:
Agreed, we can discuss this and come up with a standardized way to deal with expected output in these tests, also pinging @jsoriano here, as he is working on some ideas to improve our testing.

Reply:
The ideas I'm working on are more focused to improve and scale integration tests, but any abstraction like these ones are more than welcome of course :)